### DISCUSSION TITLE
4.1.9 -> 4.2.0 PG::InsufficientPrivilege: ERROR:  permission denied for schema public

### DISCUSSION THREAD

**Discussion by truemanner** [mastodon/mastodon#27086] [General]
i have problem with db upgrade

`SKIP_POST_DEPLOYMENT_MIGRATIONS=true RAILS_ENV=production bundle exec rails db:migrate
I, [2023-09-23T11:18:23.963739 #2321720]  INFO -- : Migrating to CreateBulkImports (20230330135507)
== 20230330135507 CreateBulkImports: migrating ================================
[strong_migrations] DANGER: No lock timeout set
-- create_table(:bulk_imports)
rails aborted!
StandardError: An error has occurred, this and all later migrations canceled:

PG::InsufficientPrivilege: ERROR:  permission denied for schema public
LINE 1: CREATE TABLE "bulk_imports" ("id" bigserial primary key, "ty...`

what to do?

**Comment by truemanner** [comment]
Can anyone help me?

**Comment by gnh1201** [comment]
I had the same problem and this article helped.

* https://stackoverflow.com/questions/74110708/postgres-15-permission-denied-for-schema-public (db permission)
* https://github.com/ankane/strong_migrations (timeout)

**Reply by truemanner** [reply]
thank you :)
only this 

"ALTER DATABASE mastodon_production OWNER TO mastodon;"

worked for me and i'am back online
